sdcm/cluster.py

```diff
@@ -4688,11 +4688,27 @@ def kill_stress_thread(self):
         self.kill_docker_loaders()
 
     def kill_cassandra_stress_thread(self):
+        search_cmds = [
+            'pgrep -f .*cassandra.*',
+            'pgrep -f cassandra.stress',
+            'pgrep -f cassandra-stress'
+        ]
+
+        def kill_cs_process(loader, filter_cmd):
+            list_of_processes = loader.remoter.run(cmd=filter_cmd,
+                                                   verbose=True, ignore_status=True)
+            if not list_of_processes.stdout.strip():
+                return True
+            loader.remoter.run(cmd=f'{filter_cmd} | xargs -I{{}}  kill -TERM {{}}',
+                               verbose=True, ignore_status=True)
+            return False
+
         for loader in self.nodes:
             try:
-                loader.remoter.run(cmd='pgrep -f cassandra.stress | xargs -I{}  kill -TERM {}',
-                                   verbose=False, ignore_status=True)
-                self.log.info("Killed cassandra stress on node: %s", loader.name)
+                for search_cmd in search_cmds:
+                    wait.wait_for(kill_cs_process, text="Search and kill c-s processes", timeout=30, throw_exc=False,
```

Comment:
I'm not sure I understand why we are doing `wait_for` here ? i.e. it would try the same command 30 times on case of failure. especially  when other tools are being used (like ycsb, ndbench, scylla-bench)

Also why it's related to this PR fix ? please open a new PR for it, with the explanation why we need this changed.

Reply:
The reason is, that c-s command continue to run, after it was killed with pgrep -f cassandra.stress | xargs -I{} kill -TERM {}. This happend on latest sct master and perf-latency with nemesis job, when we kill c-s command after first round. 
i am not sure why it happened, but after main process cassandra-stress had been killed, the java -sever with c-s command restarted and continue to run the traffic. and these cause that job aborted by timeout

i added  wait_for to be sure that if for some reason c-s was  continue to run or restarted, it will be killed again.

Reply:
i think it should n't affect on other tools

Reply:
@fruch new PR https://github.com/scylladb/scylla-cluster-tests/pull/4879